core/src/main/java/org/kiwix/kiwixmobile/core/extensions/FragmentExtensions.kt

```diff
@@ -28,9 +30,12 @@ inline fun <reified T : ViewModel> Fragment.viewModel(
   viewModelFactory: ViewModelProvider.Factory
 ) = ViewModelProviders.of(this, viewModelFactory).get(T::class.java)
 
-fun Fragment.toast(
-  stringId: Int,
-  length: Int = Toast.LENGTH_LONG
-) {
+fun Fragment.toast(stringId: Int, length: Int = Toast.LENGTH_LONG) {
   requireActivity().toast(stringId, length)
 }
+
+fun Fragment.closeKeyboard() {
```

Comment:
Have you read this infamous [SO answer](https://stackoverflow.com/a/17789187/12225251)?

`close` is quite misleading when what this method does is actually "toggle"

Reply:
It seems to always close the keyboard, if I call it twice, the keyboard is not hidden and then shown, but only hidden. If I call "toggleKeyboard" in onCreate, the keyboard is not opened. Am I missing something?

Reply:
![image](https://user-images.githubusercontent.com/3358282/93864515-ed174d80-fcbc-11ea-8c3d-2e6395de776b.png)

regardless it is all fucked up, a hide flag is being used in a show flag position and it only works because they are the same constant. We should fix it

Reply:
I've tried most of the things mentioned in the SO post and can not make it work. I think this will have to wait for the weekend  if it's to be changed. 

```kt
fun Fragment.closeKeyboard(context: Context) {
  view?.rootView?.let {
    val imm = context.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
    imm.hideSoftInputFromWindow(it.windowToken, 0)
    it.clearFocus()
  }
}
```
called like: 
```kt

  override fun onDestroyView() {
    super.onDestroyView()
    compositeDisposable.clear()
    closeKeyboard(requireContext())
  }
```

from `SearchFragment.kt`

Reply:
well I suppose it is minor and if it works it works, the release should march forward and we can accept mild debt